### PR TITLE
Support Sylius 1.7 again

### PR DESCRIPTION
**Pull request by Zales0123**
With the latest `v1.0.0-RC.4`, we've introduced support for Doctrine Migrations 3.0. However, it required from us also to support only Sylius ^1.8... with the support for Sylius ^1.5 in the previous tag.

I believe it was a mistake. Doing such a big and work-demanding change as migrations version switch together with bumping Sylius requirement by 3 versions can be a pain in the back for a lot of developers using `RefundPlugin` already.

I think that doing a step backward can improve the developer experience for those, who already use this plugin. Definitely, the next release (hopefully the stable one already 🤞) would support Sylius 1.8 only, so this one is just an intermediate stage. The implementation is not perfect as well but allows us to test 2 versions of Sylius along with 2 versions of PHP and MySQL 💃 

I would love to see your opinion 🖖

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -91,9 +92,35 @@ jobs:
...
                         ${{ runner.os }}-php-${{ matrix.php }}-composer-
 
+            -
+                name: Prepare migrations for Sylius 1.7
+                if: matrix.sylius == '1.7.*'
+                run: |
+                    cp src/Migrations/* tests/Application/src/Migrations/
+                    cd tests/Application/src/Migrations
+                    sed -i -- 's/Sylius\\RefundPlugin\\Migrations/DoctrineMigrations/g' *
+                    cd ../../../../
```
Comment (oallain) [inline .github/workflows/build.yml:102]: Can be simplified with :
```suggestion
                    sed -i -- 's/Sylius\\RefundPlugin\\Migrations/DoctrineMigrations/g' tests/Application/src/Migrations/*
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -91,9 +92,35 @@ jobs:
...
+                if: matrix.sylius == '1.8.*'
+                run: |
+                    composer require sylius-labs/doctrine-migrations-extra-bundle --no-interaction --no-update
+                    cp etc/travis/sylius18/bundles.php tests/Application/config/
+                    cp etc/travis/sylius18/doctrine_migrations.yaml tests/Application/config/packages/
+                    cp etc/travis/sylius18/lexik_jwt_authentication.yaml tests/Application/config/packages/
+                    cp etc/travis/sylius18/security.yaml tests/Application/config/packages/
+                    cp etc/travis/sylius18/sylius_api.yaml tests/Application/config/routes/
+                    rm tests/Application/src/Migrations/Version*
+                    cat tests/Application/config/packages/doctrine_migrations.yaml
```
Comment (oallain) [inline .github/workflows/build.yml:114]:
```suggestion
```

File: tests/Application/src/Migrations/Version20200122082429.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('CREATE TABLE sylius_channel_countries (channel_id INT NOT NULL, country_id INT NOT NULL, INDEX IDX_D96E51AE72F5A1AA (channel_id), INDEX IDX_D96E51AEF92F3E70 (country_id), PRIMARY KEY(channel_id, country_id)) DEFAULT CHARACTER SET UTF8 COLLATE `UTF8_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE sylius_channel_countries ADD CONSTRAINT FK_D96E51AE72F5A1AA FOREIGN KEY (channel_id) REFERENCES sylius_channel (id) ON DELETE CASCADE');
+        $this->addSql('ALTER TABLE sylius_channel_countries ADD CONSTRAINT FK_D96E51AEF92F3E70 FOREIGN KEY (country_id) REFERENCES sylius_country (id) ON DELETE CASCADE');
```
Comment (oallain) [inline tests/Application/src/Migrations/Version20200122082429.php:27]: Tips : You can use `doctrine:migrations:diff --formatted` to format the generated SQL.

**Review by oallain — APPROVED** [review]
Hi @Zales0123 

It's great, and I thank you for that :rocket: 
I put minor comments, but this PR can be merged without if needed.

**Comment by diimpp** [comment]
On off topic note, I hope stable release won't be done until architectural problems are solve as discusses in https://github.com/Sylius/RefundPlugin/issues/193

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -91,9 +92,35 @@ jobs:
...
+                    cp src/Migrations/* tests/Application/src/Migrations/
+                    cd tests/Application/src/Migrations
+                    sed -i -- 's/Sylius\\RefundPlugin\\Migrations/DoctrineMigrations/g' *
+                    cd ../../../../
+
+            -   name: Prepare migrations for Sylius 1.8
+                if: matrix.sylius == '1.8.*'
+                run: |
+                    composer require sylius-labs/doctrine-migrations-extra-bundle --no-interaction --no-update
+                    cp etc/travis/sylius18/bundles.php tests/Application/config/
```
Comment (GSadee) [inline .github/workflows/build.yml:108]: I wonder if it shouldn't be done the other way around, to have test application configured for 1.8 and prepared configs for 1.7 in `etc/` directory

File: etc/travis/sylius18/bundles.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
+<?php
```
Comment (GSadee) [inline etc/travis/sylius18/bundles.php:1]: These configs should not be in `etc/travis` directory as we don't use travis anymore on this repository

File: .github/workflows/build.yml (hunk 4 above)
Comment (Zales0123) [reply .github/workflows/build.yml:108]: It makes sense 🖖 :D

**Review by Tomanhez — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Review by oallain — APPROVED** [review]

**Comment by alopez-pp** [comment]
Hi, PayPlug team here 👋 

We are working closely with Synolia in order to release our first Sylius x PayPlug plugin, we are pretty excited to make it happen! 🚀 

I have two questions for you:
- When will you tag this modification ? 
- Regarding your message "Definitely, the next release (hopefully the stable one already 🤞) would support Sylius 1.8 only", Does this mean that there is no hope at all to get the refund plugin in a stable version for 1.7 ? Can you explain why ? 

Thanks !

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Mateusz! :1st_place_medal:

**Comment by Zales0123** [comment]
Hi @alopez-pp!

Thank you for your kind message :) As the PR is merged we will release the next tag as soon as possible. 

Regarding the second question. Sylius 1.7 is already not actively maintained for the last 2 weeks, it will get only security fixes for the next months. It's highly recommended to everyone to upgrade to the latest supported 1.8 and, as creators of both Sylius and this plugin, we need to bump to supported releases as fast as possible. I think you understand our perspective 🖖 

If you have any more questions, I would be glad to answer them 🚀

**Pull request merged**